### PR TITLE
Completing the AsyncAPI model with add the schema related part

### PR DESCRIPTION
**Pull request by lbroudoux**
+ Adding commands for channels

As required for the implementation of a fully featured AsyncAPI graphical editor into Apicurio Studio, the model as been completed with all the details related to `Schema` and `SchemaDefinition` management. 

I also started to add new commands in order to add/remove `Channels` into an AyncAPI document model.

### CODE REVIEW

File: src/main/java/io/apicurio/datamodels/asyncapi/v2/io/Aai20DataModelReader.java [hunk 1]
```diff
@@ -1,14 +1,64 @@
...
+        
+        // components
+        /*
+        Object json_ = JsonCompat.consumeProperty(json, Constants.PROP_COMPONENTS);
+        if (json_ != null) {
+            AaiComponents components = nodeFactory.createComponents(node);
+            this.readComponents(json_, components);
+            doc.components = components;
+        }
+        */
```
Comment (EricWittmann) [inline src/main/java/io/apicurio/datamodels/asyncapi/v2/io/Aai20DataModelReader.java:42]: Can probably remove this.

**Comment by EricWittmann** [comment]
This looks quite good to me.  One thing to perhaps discuss before merging is that we didn't implement Schema for AsyncAPI because the spec indicated (at the time) that multiple schema languages would be supported.  So we were concerned that implementing it the way we did for OpenAPI (which specifies the schema language format) would not work well.

Has this been changed in the spec?  I'm worried about making sure the data model supports other schema languages like Avro and Protobuf.

**Comment by lbroudoux** [comment]
Understand the concern. I thought I also read this once but now it is stated in the specification that:

"The AsyncAPI Schema Object is a JSON Schema vocabulary which extends JSON Schema Core and Validation vocabularies." 

Also:

"This object is a superset of the JSON Schema Specification Draft 07."

Maybe other schema formats could be managed as extension or external references? But it seems that the core part is fully JSON Schéma compliant... From my understanding at least 😌

File: src/main/java/io/apicurio/datamodels/asyncapi/v2/io/Aai20DataModelReader.java (hunk 1 above)
Comment (lbroudoux) [reply src/main/java/io/apicurio/datamodels/asyncapi/v2/io/Aai20DataModelReader.java:42]: Yep for sure.

**Review by EricWittmann — APPROVED** [review]

**Pull request merged**